### ISSUE TITLE
Self update for youtube-dl

### ISSUE THREAD

**Issue by Rpsl** [mxpv/podsync#119]
I think can add additional cron job `@daily` for execution self update of youtube-dl
```
$ youtube-dl -U 
youtube-dl is up-to-date (2020.03.24)
```

**Comment by 726a67** [comment]
Agreed, this can particularly frustrate usage via docker, as users pulling the latest image from dockerhub are limited to the version of youtube-dl explicitly installed by the current Dockerfile (2020.01.24).

A cron job used to regularly self-update youtube-dl could solve this problem, but it might also be worth considering the use of pip to install the latest version of youtube-dl at image build time.

**Comment by mxpv** [comment]
Can you guys give it a try? (you can download latest non-release binaries from here: https://github.com/mxpv/podsync/actions/runs/82314116)

**Comment by 726a67** [comment]
@mxpv Could you upload/tag a 'preview'/'test' release to dockerhub using the latest non-release binary? I only deploy podsync via docker, so testing the binary directly is problematic.

**Comment by mxpv** [comment]
Will add: #135

**Comment by DeXteRrBDN** [comment]
Managed to work with current Docker version and latest youtube-dl, just added the following mapping to docker-compose.yml:

```
version: '2.2'

services:
  podsync:
    container_name: podsync
    image: mxpv/podsync:latest
    restart: always
    ports:
      - 80:80
    volumes:
      - ./data:/app/data/
      - ./config.toml:/app/config.toml
      **- ./youtube-dl:/usr/bin/youtube-dl**
```

Then downloaded youtube-dl from latest version on GitHub and it started to work. I can update youtube-dl manually now using docker image!

**Comment by mxpv** [comment]
Self update is implemented in 2.3.0. 
Use the following in your config file:
```
[downloader]
self_update = true # Optional, auto update youtube-dl every 24 hours
```
